File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/General/MyExceptionHandler.java
package kore.botssdk.audiocodes.webrtcclient.General;


import android.content.Context;
import android.content.Intent;

import java.io.PrintWriter;
import java.io.StringWriter;

public class MyExceptionHandler implements
        Thread.UncaughtExceptionHandler {
    private final Context myContext;
    private final Class<?> myActivityClass;

    public MyExceptionHandler(Context context, Class<?> c) {
        myContext = context;
        myActivityClass = c;
    }

    public void uncaughtException(Thread thread, Throwable exception) {
        Log.e("MyExceptionHandler","uncaughtException: "+exception.toString());
        StringWriter stackTrace = new StringWriter();
        exception.printStackTrace(new PrintWriter(stackTrace));
        System.err.println(stackTrace);// You can use LogCat too
        Intent intent = new Intent(myContext, myActivityClass);
        String s = stackTrace.toString();
        //you can use this String to know what caused the exception and in which Activity
        intent.putExtra("uncaughtException", "Exception is: " + stackTrace.toString());
        intent.putExtra("stacktrace", s);
        myContext.startActivity(intent);
        //for restarting the Activity
        //Process.killProcess(Process.myPid());
        android.os.Process.killProcess(android.os.Process.myPid());
        System.exit(0);
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/general/MyExceptionHandler.kt
package com.kore.ui.audiocodes.webrtcclient.general

import android.content.Context
import android.content.Intent
import android.os.Process
import com.kore.ui.audiocodes.webrtcclient.general.Log.e
import java.io.PrintWriter
import java.io.StringWriter

class MyExceptionHandler(private val myContext: Context, private val myActivityClass: Class<*>) : Thread.UncaughtExceptionHandler {
    override fun uncaughtException(thread: Thread, exception: Throwable) {
        e("MyExceptionHandler", "uncaughtException: $exception")
        val stackTrace = StringWriter()
        exception.printStackTrace(PrintWriter(stackTrace))
        System.err.println(stackTrace) // You can use LogCat too
        val intent = Intent(myContext, myActivityClass)
        val s = stackTrace.toString()
        //you can use this String to know what caused the exception and in which Activity
        intent.putExtra("uncaughtException", "Exception is: $stackTrace")
        intent.putExtra("stacktrace", s)
        myContext.startActivity(intent)
        //for restarting the Activity
        //Process.killProcess(Process.myPid());
        Process.killProcess(Process.myPid())
        System.exit(0)
    }
}
